detect_secrets/core/bidirectional_iterator.py

```diff
@@ -0,0 +1,31 @@
+class BidirectionalIterator(object):
+    def __init__(self, collection):
+        self.collection = collection
+        self.index = -1 #starts on -1, as index is increased _before_ getting result
+        self.step_back_once = False 
+
+    def __next__(self):
+        try:
+            if self.step_back_once:
+                self.index -= 1
+                self.step_back_once = False 
+                if self.index < 0:
+                    raise StopIteration
+            else:
+                self.index += 1
+            result = self.collection[self.index]
+        except IndexError:
+            raise StopIteration
+        return result
+
+    def next(self):
```

Comment:
Nit: Looping should call the `__next__` method directly, so no need for a `next` method 👍

Reply:
I added `next` to be python2 compatible (it was renamed form `next` to `__next__` from python2 to 3).
However, this does not feel very clean - perhaps you know of a better way? :)

Reply:
Aha! Very good point, I only tested on Python 3. My bad. I am impressed by how thorough you are 👍